### ISSUE TITLE
False positive for [MD011] “Reversed link syntax”

### ISSUE THREAD

**Issue by jan-ferdinand** [joshrotenberg/mdbook-lint#322]
Consider the following section in an mdbook:

```md
## Title

[link](#title)[^footnote]

[^footnote]: my footnote
```

Running mdbook-lint incorrectly triggers [MD011], presumably because of the `(#title)[^footnote]` pairing. However, the correct pairing is `[link](#title)`; the `[^footnote]` should then be considered separately.
